pkg/oomstore/revision.go

```diff
@@ -24,6 +25,50 @@ func (s *OomStore) GetRevisionBy(ctx context.Context, groupID int, revision int6
 	return s.metadata.GetRevisionBy(ctx, groupID, revision)
 }
 
+func (s *OomStore) CreateRevision(ctx context.Context, opt metadata.CreateRevisionOpt) (int, string, error) {
+	var (
+		revisionID    int
+		snapshotTable string
+		err           error
+
+		domainRevision *types.Revision
+	)
```

Comment:
Can you define the variables near to it's uage if possible? It's hard to follow what the  value was for each variable and when it was used in this style.

Reply:
These variables will be assigned values in the WithTransaction and can only be defined outside.
Another way to write it is:

```go
var revisionID int
var snapshotTable string
var err error
var domainRevision *types.Revision
```

Reply:
Ok. err is not nesessary pre decleared?
```go
var revisionID int
var snapshotTable string
var domainRevision *types.Revision

if err := ...
```

Reply:
err can be defined where it is used, I'll change it later. 

```
var revisionID int
var snapshotTable string
var domainRevision *types.Revision
```
is better than?

```
var (
    revisionID int
    snapshotTable string
    domainRevision *types.Revision
)
```

Reply:
I prefer the first but it's ok if you insist the second.